src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -51,6 +61,23 @@ def get_observation_histograms(self):
         """
         return self._scheduler.get_observation_histograms()
 
+    def get_expanded_specs(self) -> Dict[str, List[str]]:
+        """Return a dict containing the canonicalized addresses of the specs for this run, and what
+        files they expand to."""
+
+        (addresses,) = self._scheduler.product_request(
+            Addresses, [Params(self._specs, self._options_bootstrapper)]
+        )
+
+        ret = {}
+        for addr in addresses:
+            (targets,) = self._scheduler.product_request(Targets, [Params(Addresses([addr]))])
```

Comment:
Also, why do this? You don't use any feature from `Targets` - I'm assuming it's so that you can "expand" a spec like `src/python/pants/util:util` to include all of its corresponding file targets.

Reply:
> Also, why do this? You don't use any feature from `Targets` - I'm assuming it's so that you can "expand" a spec like `src/python/pants/util:util` to include all of its corresponding file targets.

That's exactly the idea. If there's another way to do this, I'd be interested.

Reply:
Cool, that's correct then. I suggest renaming things to express that, e.g. `unexpanded_addr` and `expanded_targets`.

Reply:
This is already a batch API:
```
targets_per_address = self._scheduler.product_request(Targets, [Params(Addresses([addr])) for addr in addresses])
```